packages/select-menu/src/types.ts

```diff
@@ -47,16 +47,11 @@ export interface SelectMenuProps<T extends OptionBase> {
   onChange: (value: string | string[]) => void
   onSelect: (value: string) => void
   onDeselect: (value: string) => void
-  multiOptionSelectedItemsLabel?: (itemsSelectedLength: number) => string
-  isMultiSelect?: boolean
-  // compareBy?: string
-  placeholder?: string
-  // renderItem?: (
-  //   option: T,
-  //   click: () => void,
-  //   selected: boolean,
-  //   options: T[]
-  // ) => React.ReactNode
+  placeholder?: string | ((selected: T[]) => React.ReactNode)
```

Comment:
Actually if we're using `array` vs `string?` differentiation with `value` prop for single-select/multi-select, we can ditch one of placeholder props now - e.g. placeholderForMultipleSelected could be dropped

Reply:
```tsx
<SelectMenu value={[]} placeholder={selected => selected.length === 0 ? 'Select...' : selected.length === 1 ? selected[0].label : `Tags: ${selected.length}`} />
// is the same as
<SelectMenu value={[]} placeholderForMultipleSelected={selected => `Tags: ${selected.length}`} />
```

`placeholderForMultipleSelected` is not the best name but it's useful for customizing placeholder in case when there are more than one element selected

Reply:
Heck, I've invented it myself (`multiOptionSelectedItemsLabel`) and now I think it's reduntant 😿 
In this case, it's vital to add proper documentation for both `placeholder` and `placeholderForMultipleSelected` - AFAIK first prop is undocumented

Reply:
I don't have any better name than `placeholder`.  `label` will be confused with real label that displays before component. 

Thanks for docs 👍